File: GdPicture/WinForm/C#/Barcode Recognition/BarcodeResults.cs
using System.Collections.Generic;

namespace barcode_recognition
{

    public partial class BarcodeResults
    {

        private List<SingleBarcodeInfo> _barcode1DResults;
        private List<SingleBarcodeInfo> _barcode2DResults;
        private List<SingleBarcodeInfo> _barcodeQRResults;
        private List<SingleBarcodeInfo> _barcodePDF417Results;
        private List<SingleBarcodeInfo> _barcodeMicroQRResults;
        private List<SingleBarcodeInfo> _barcodeAztecResults;
        private List<SingleBarcodeInfo> _barcodeMaxicodeResults;
        private List<SingleBarcodeInfo> _patchCodeResults;

        private int _totalBarcodesCount = 0;

        public BarcodeResults(List<SingleBarcodeInfo> barcode1DResults, List<SingleBarcodeInfo> barcode2DResults,
            List<SingleBarcodeInfo> barcodeQrResults, List<SingleBarcodeInfo> barcodePDF417Results, List<SingleBarcodeInfo> barcodeMicroQrResults,
            List<SingleBarcodeInfo> barcodeAztecResults, List<SingleBarcodeInfo> barcodeMaxicodeResults, List<SingleBarcodeInfo> patchCodeResults)
        {
            InitializeComponent();

            this._barcode1DResults = barcode1DResults;
            this._barcode2DResults = barcode2DResults;
            this._barcodeQRResults = barcodeQrResults;
            this._barcodePDF417Results = barcodePDF417Results;
            this._barcodeMicroQRResults = barcodeMicroQrResults;
            this._barcodeAztecResults = barcodeAztecResults;
            this._patchCodeResults = patchCodeResults;
            this._barcodeMaxicodeResults = barcodeMaxicodeResults;

            EnableButtons();
        }

        private void EnableButtons()
        {
            bool populated = false;
            if (this._barcode1DResults != null && this._barcode1DResults.Count > 0)
            {
                this.btn1DBarcode.Text = this.btn1DBarcode.Text + ": " + this._barcode1DResults.Count;
                _totalBarcodesCount += this._barcode1DResults.Count;
                this.btn1DBarcode.Enabled = true;
                PopulateList(this._barcode1DResults);
                populated = true;
            }

            if (this._barcode2DResults != null && this._barcode2DResults.Count > 0)
            {
                this.btn2DBarcode.Text = this.btn2DBarcode.Text + ": " + this._barcode2DResults.Count;
                _totalBarcodesCount += this._barcode2DResults.Count;
                this.btn2DBarcode.Enabled = true;
                if (!populated)
                {
                    PopulateList(this._barcode2DResults);
                    populated = true;
                }
            }

            if (this._barcodeQRResults != null && this._barcodeQRResults.Count > 0)
            {
                this.btnQRBarcode.Text = this.btnQRBarcode.Text + ": " + this._barcodeQRResults.Count;
                _totalBarcodesCount += this._barcodeQRResults.Count;
                this.btnQRBarcode.Enabled = true;
                if (!populated)
                {
                    PopulateList(this._barcodeQRResults);
                    populated = true;
                }
            }

            if (this._barcodePDF417Results != null && this._barcodePDF417Results.Count > 0)
            {
                this.btnPDF417Barcode.Text = this.btnPDF417Barcode.Text + ": " + this._barcodePDF417Results.Count;
                _totalBarcodesCount += this._barcodePDF417Results.Count;
                this.btnPDF417Barcode.Enabled = true;
                if (!populated)
                {
                    PopulateList(this._barcodePDF417Results);
                    populated = true;
                }
            }

            if (this._barcodeMicroQRResults != null && this._barcodeMicroQRResults.Count > 0)
            {
                this.btnMicroQRBarcode.Text = this.btnMicroQRBarcode.Text + ": " + this._barcodeMicroQRResults.Count;
                _totalBarcodesCount += this._barcodeMicroQRResults.Count;
                this.btnMicroQRBarcode.Enabled = true;
                if (!populated)
                {
                    PopulateList(this._barcodeMicroQRResults);
                    populated = true;
                }
            }

            if (this._barcodeAztecResults != null && this._barcodeAztecResults.Count > 0)
            {
                this.btnAztecBarcode.Text = this.btnAztecBarcode.Text + ": " + this._barcodeAztecResults.Count;
                _totalBarcodesCount += this._barcodeAztecResults.Count;
                this.btnAztecBarcode.Enabled = true;
                if (!populated)
                {
                    PopulateList(this._barcodeAztecResults);
                    populated = true;
                }
            }

            if (this._barcodeMaxicodeResults != null && this._barcodeMaxicodeResults.Count > 0)
            {
                this.btnMaxicodeBarcode.Text = this.btnMaxicodeBarcode.Text + ": " + this._barcodeMaxicodeResults.Count;
                _totalBarcodesCount += this._barcodeMaxicodeResults.Count;
                this.btnMaxicodeBarcode.Enabled = true;
                if (!populated)
                {
                    PopulateList(this._barcodeMaxicodeResults);
                    populated = true;
                }
            }

            if (this._patchCodeResults != null && this._patchCodeResults.Count > 0)
            {
                this.btnPatchCodeBarcode.Text = this.btnPatchCodeBarcode.Text + ": " + this._patchCodeResults.Count;
                _totalBarcodesCount += this._patchCodeResults.Count;
                this.btnPatchCodeBarcode.Enabled = true;
                if (!populated)
                {
                    PopulateList(this._patchCodeResults);
                    populated = true;
                }
            }

            lblTotalBarcodesCount.Text = lblTotalBarcodesCount.Text + " " + _totalBarcodesCount;
        }

        private void Close_Click(object sender, System.EventArgs e)
        {
            Close();
        }

        private void Form_Closing(object sender, System.Windows.Forms.FormClosingEventArgs e)
        {
            ClearLists();
        }

        private void ClearLists()
        {
            _barcode1DResults.Clear();
            _barcode2DResults.Clear();
            _barcodeQRResults.Clear();
            _barcodePDF417Results.Clear();
            _barcodeMicroQRResults.Clear();
            _barcodeAztecResults.Clear();
            _barcodeMaxicodeResults.Clear();
            _patchCodeResults.Clear();
        }

        private void PopulateList(List<SingleBarcodeInfo> barcodesResultsList)
        {
            resultsList.Controls.Clear();
            for (int i = 0; i <= barcodesResultsList.Count - 1; i++)
            {
                resultsList.Controls.Add(barcodesResultsList[i]);
            }
        }

        private void btn1DBarcode_Click(object sender, System.EventArgs e)
        {
            PopulateList(this._barcode1DResults);
        }

        private void btn2DBarcode_Click(object sender, System.EventArgs e)
        {
            PopulateList(this._barcode2DResults);
        }

        private void btnQRBarcode_Click(object sender, System.EventArgs e)
        {
            PopulateList(this._barcodeQRResults);
        }

        private void btnPDF417Barcode_Click(object sender, System.EventArgs e)
        {
            PopulateList(this._barcodePDF417Results);
        }

        private void btnMicroQRBarcode_Click(object sender, System.EventArgs e)
        {
            PopulateList(this._barcodeMicroQRResults);
        }

        private void btnPatchCodeBarcode_Click(object sender, System.EventArgs e)
        {
            PopulateList(this._patchCodeResults);
        }

        private void btnAztecBarcode_Click(object sender, System.EventArgs e)
        {
            PopulateList(this._barcodeAztecResults);
        }

        private void btnMaxicodeBarcode_Click(object sender, System.EventArgs e)
        {
            PopulateList(this._barcodeMaxicodeResults);
        }
    }
}

File: GdPicture/WinForm/vb.net/Barcode Recognition/BarcodeResults.vb
Imports System.Collections.Generic

Public Class BarcodeResults

    Private _barcode1DResults As List(Of SingleBarcodeInfo)
    Private _barcode2DResults As List(Of SingleBarcodeInfo)
    Private _barcodeQRResults As List(Of SingleBarcodeInfo)
    Private _barcodePDF417Results As List(Of SingleBarcodeInfo)
    Private _barcodeMicroQRResults As List(Of SingleBarcodeInfo)
    Private _barcodeAztecResults As List(Of SingleBarcodeInfo)
    Private _barcodeMaxicodeResults As List(Of SingleBarcodeInfo)
    Private _patchCodeResults As List(Of SingleBarcodeInfo)

    Private totalBarcodesCount As Integer = 0

    Public Sub New(ByVal barcode1DResults As List(Of SingleBarcodeInfo),
                   ByVal barcode2DResults As List(Of SingleBarcodeInfo),
                   ByVal barcodeQRResults As List(Of SingleBarcodeInfo),
                   ByVal barcodePDF417Results As List(Of SingleBarcodeInfo),
                   ByVal barcodeMicroQRResults As List(Of SingleBarcodeInfo),
                   ByVal barcodeAztecResults As List(Of SingleBarcodeInfo),
                   ByVal barcodeMaxicodeResults As List(Of SingleBarcodeInfo),
                   ByVal PatchCodeResults As List(Of SingleBarcodeInfo))
        InitializeComponent()

        Me._barcode1DResults = barcode1DResults
        Me._barcode2DResults = barcode2DResults
        Me._barcodeQRResults = barcodeQRResults
        Me._barcodePDF417Results = barcodePDF417Results
        Me._barcodeMicroQRResults = barcodeMicroQRResults
        Me._barcodeAztecResults = barcodeAztecResults
        Me._barcodeMaxicodeResults = barcodeMaxicodeResults
        Me._patchCodeResults = PatchCodeResults

        EnableButtons()
    End Sub

    Private Sub EnableButtons()
        Dim populated As Boolean = False
        If Me._barcode1DResults IsNot Nothing AndAlso Me._barcode1DResults.Count > 0 Then
            Me.btn1DBarcode.Text = Me.btn1DBarcode.Text & ": " & Me._barcode1DResults.Count.ToString()
            totalBarcodesCount += Me._barcode1DResults.Count
            Me.btn1DBarcode.Enabled = True
            PopulateList(Me._barcode1DResults)
            populated = True
        End If

        If Me._barcode2DResults IsNot Nothing AndAlso Me._barcode2DResults.Count > 0 Then
            Me.btn2DBarcode.Text = Me.btn2DBarcode.Text & ": " & Me._barcode2DResults.Count.ToString()
            totalBarcodesCount += Me._barcode2DResults.Count
            Me.btn2DBarcode.Enabled = True
            If Not populated Then
                PopulateList(Me._barcode2DResults)
                populated = True
            End If
        End If

        If Me._barcodeQRResults IsNot Nothing AndAlso Me._barcodeQRResults.Count > 0 Then
            Me.btnQRBarcode.Text = Me.btnQRBarcode.Text & ": " & Me._barcodeQRResults.Count.ToString()
            totalBarcodesCount += Me._barcodeQRResults.Count
            Me.btnQRBarcode.Enabled = True
            If Not populated Then
                PopulateList(Me._barcodeQRResults)
                populated = True
            End If
        End If

        If Me._barcodePDF417Results IsNot Nothing AndAlso Me._barcodePDF417Results.Count > 0 Then
            Me.btnPDF417Barcode.Text = Me.btnPDF417Barcode.Text & ": " & Me._barcodePDF417Results.Count.ToString()
            totalBarcodesCount += Me._barcodePDF417Results.Count
            Me.btnPDF417Barcode.Enabled = True
            If Not populated Then
                PopulateList(Me._barcodePDF417Results)
                populated = True
            End If
        End If

        If Me._barcodeMicroQRResults IsNot Nothing AndAlso Me._barcodeMicroQRResults.Count > 0 Then
            Me.btnMicroQRBarcode.Text = Me.btnMicroQRBarcode.Text & ": " & Me._barcodeMicroQRResults.Count.ToString()
            totalBarcodesCount += Me._barcodeMicroQRResults.Count
            Me.btnMicroQRBarcode.Enabled = True
            If Not populated Then
                PopulateList(Me._barcodeMicroQRResults)
                populated = True
            End If
        End If

        If Me._barcodeAztecResults IsNot Nothing AndAlso Me._barcodeAztecResults.Count > 0 Then
            Me.btnAztecBarcode.Text = Me.btnAztecBarcode.Text & ": " & Me._barcodeAztecResults.Count.ToString()
            totalBarcodesCount += Me._barcodeAztecResults.Count
            Me.btnAztecBarcode.Enabled = True
            If Not populated Then
                PopulateList(Me._barcodeAztecResults)
                populated = True
            End If
        End If

        If Me._barcodeMaxicodeResults IsNot Nothing AndAlso Me._barcodeMaxicodeResults.Count > 0 Then
            Me.btnMaxicodeBarcode.Text = Me.btnMaxicodeBarcode.Text & ": " & Me._barcodeMaxicodeResults.Count.ToString()
            totalBarcodesCount += Me._barcodeMaxicodeResults.Count
            Me.btnMaxicodeBarcode.Enabled = True
            If Not populated Then
                PopulateList(Me._barcodeMaxicodeResults)
                populated = True
            End If
        End If

        If Me._patchCodeResults IsNot Nothing AndAlso Me._patchCodeResults.Count > 0 Then
            Me.btnPatchCodeBarcode.Text = Me.btnPatchCodeBarcode.Text & ": " & Me._patchCodeResults.Count.ToString()
            totalBarcodesCount += Me._patchCodeResults.Count
            Me.btnPatchCodeBarcode.Enabled = True
            If Not populated Then
                PopulateList(Me._patchCodeResults)
                populated = True
            End If
        End If

        lblTotalBarcodesCount.Text = lblTotalBarcodesCount.Text & " " & totalBarcodesCount.ToString()
    End Sub

    Private Sub Close_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
        Close()
    End Sub

    Private Sub Form_Closing() Handles Me.FormClosing
        ClearLists()
    End Sub

    Private Sub ClearLists()
        _barcode1DResults.Clear()
        _barcode2DResults.Clear()
        _barcodeQRResults.Clear()
        _barcodePDF417Results.Clear()
        _barcodeMicroQRResults.Clear()
        _barcodeAztecResults.Clear()
        _barcodeMaxicodeResults.Clear()
        _patchCodeResults.Clear()
    End Sub

    Private Sub PopulateList(ByVal barcodesResultsList As List(Of SingleBarcodeInfo))
        resultsList.Controls.Clear()
        For i As Integer = 0 To barcodesResultsList.Count - 1
            resultsList.Controls.Add(barcodesResultsList(i))
        Next
    End Sub

    Private Sub btn1DBarcode_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btn1DBarcode.Click
        PopulateList(Me._barcode1DResults)
    End Sub

    Private Sub btn2DBarcode_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btn2DBarcode.Click
        PopulateList(Me._barcode2DResults)
    End Sub

    Private Sub btnQRBarcode_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnQRBarcode.Click
        PopulateList(Me._barcodeQRResults)
    End Sub

    Private Sub btnPDF417Barcode_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnPDF417Barcode.Click
        PopulateList(Me._barcodePDF417Results)
    End Sub

    Private Sub btnMicroQRBarcode_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnMicroQRBarcode.Click
        PopulateList(Me._barcodeMicroQRResults)
    End Sub

    Private Sub btnPatchCodeBarcode_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnPatchCodeBarcode.Click
        PopulateList(Me._patchCodeResults)
    End Sub

    Private Sub btnAztecBarcode_Click(sender As Object, e As EventArgs) Handles btnAztecBarcode.Click
        PopulateList(Me._barcodeAztecResults)
    End Sub

    Private Sub btnMaxicodeBarcode_Click(sender As Object, e As EventArgs) Handles btnMaxicodeBarcode.Click
        PopulateList(Me._barcodeMaxicodeResults)
    End Sub
End Class